pkg/server/interfaces/api/repository.go

```diff
@@ -131,6 +133,28 @@ func (h repository) GetWebServiceRoute() *restful.WebService {
 		Returns(400, "Bad Request", bcode.Bcode{}).
 		Writes([]string{}))
 
+	ws.Route(ws.POST("/registrysecrets/verify").To(h.verifyRepositorySecret).
```

Comment:
Why do we need this API?

Reply:
verifies image secret against registry, it takes k8s.io/api/core/v1/types.Secret as input, and authenticate registry with credential specified. Returns http.StatusOK if authenticate successfully, returns http.StatusUnauthorized if failed.

Reply:
Checking the registry config when using it is not a good idea. The best way is that verifies it when creating the config. 
Provide a solution, adding a provider to verify the image registry, and use it in the config template.

Reply:
We can add a verification button when creating the config and then call the api to do the verification